src/backlight/portfolio/portfolio.py

```diff
@@ -18,21 +19,25 @@ class Portfolio:
     Each element of the list represent the positions of an asset that contribute to the portfolio
     """
 
-    def __init__(self, positions: List[Positions]):
+    _metadata = ["currency_unit"]
+
+    def __init__(
+        self, positions: List[Positions], currency_unit: Currency = Currency.USD
+    ):
         """
         If there is positions with the same symbol, their value are sum
         """
         symbols = [position.symbol for position in positions]
         assert len(symbols) == len(set(symbols))
         self._positions = positions
 
-    def value(self) -> pd.DataFrame:
+    @property
+    def value(self) -> pd.Series:
         """ DataFrame of the portfolio valuation of each asset"""
         pl = pd.DataFrame()
         for p in self._positions:
-            # Compute PL of positions of each asset
-            pl[p.symbol] = calc_pl(p)
-        return pl
+            pl[p.symbol] = p.value
+        return pl.sum(axis=1)
```

Comment:
```
def value(self) -> pd.Series:
    values = [p.value for p in self._positions]
    return values.sum(axis=1)
```

Which one is faster?

Reply:
I'll compare both.

Reply:
Your solution is way faster. I add it to the code.